docs/MatrixOne/Develop/schema-design/1.1-vector.md

```diff
@@ -0,0 +1,176 @@
+
+- **Dimensional Flexibility:** Vector capabilities typically support vectors of different dimensions, allowing the database to handle various data types and features.
+
+- **Integration with Machine Learning:** Databases that support vector storage and querying can integrate seamlessly with machine learning models, making model training and inference more efficient and convenient.
+
+## Before you start
+
+Before reading this document, make sure that the following tasks are completed:
+
+- Build a MatrixOne Cluster in MatrixOne.
+- Read the [Database Schema Design Overview](overview.md).
+- The database has been created.
+
+## How to Use Vectors
+
+The syntax for using vectors is the same as that for regular table creation, data insertion, and data querying:
+
+### Creating
+
+Using the following SQL statement, you can create two vector columns, one of type Float32 and the other of type Float64. You can also set the dimensions for both vector columns to 3.
+
+```
+create table t1(a int, b vecf32(3), c vecf64(3));
+```
+
+### Inserting
+
+MatrixOne supports inserting vectors in two formats.
+
+**Textual Format**
+
+```sql
+insert into t1 values(1, "[1,2,3]", "[4,5,6]");
+```
+
+**Binary Format**
+
+Suppose you are working with Python NumPy arrays. In that case, you can directly insert the NumPy array into MatrixOne by performing hexadecimal encoding on the array instead of converting it into a comma-separated textual format. This approach is faster when inserting vectors with higher dimensions.
+
+```sql
+insert into t1 (a, b) values
+(2, decode("7e98b23e9e10383b2f41133f", "hex"));
+
+-- "7e98b23e9e10383b2f41133f" represents the hexadecimal encoding of the little-endian []float32{0.34881967, 0.0028086076, 0.5752134}
+
+-- "hex" represents hexadecimal encoding
+```
+
+### Querying
+
+Vector columns can also be read in two formats.
+
+**Textual Format**
+
+```sql
+mysql> select a, b from t1;
++------+---------------------------------------+
+| a    | b                                     |
++------+---------------------------------------+
+|    1 | [1, 2, 3]                             |
+|    2 | [0.34881967, 0.0028086076, 0.5752134] |
++------+---------------------------------------+
+2 rows in set (0.00 sec)
+```
+
+**Binary Format**
+
+The binary format is very useful if you need to directly read the vector result set into a NumPy array with minimal conversion cost.
+
+```sql
+mysql> select encode(b, "hex") from t1;
++--------------------------+
+| encode(b, hex)           |
++--------------------------+
+| 0000803f0000004000004040 |
+| 7e98b23e9e10383b2f41133f |
++--------------------------+
+2 rows in set (0.00 sec)
+```
+
+## Example - Top K Queries
+
+Top K queries are a database query operation that retrieves the top K data items or records from a database. Top K queries can be applied in various scenarios, including recommendation systems, search engines, data analysis and sorting.
```

Comment:
Top K queries are a database query operation that retrieves the top K data items or records from a database. These queries find application in a wide variety of scenarios, including recommendation systems, search engines, and data analysis, among other fields.